### PR TITLE
Add kubernetes-dns-node-cache-1.22

### PR DESCRIPTION
**Pull request by Dentrax**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0
- [ ] Patch source: _patch source here_

### CODE REVIEW

File: kubernetes-dns-node-cache-1.22.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  - uses: go/build
+    with:
+      packages: ./cmd/node-cache
+      output: node-cache
+      ldflags: -s -w -X github.com/kubernetes/dns/pkg/version.Version=v${{package.version}}
+
+  - uses: strip
+
+subpackages:
+  - name: kubernetes-csi-node-driver-registrar-compat
```
Comment (developer-guy) [inline kubernetes-dns-node-cache-1.22.yaml:41]: should we add version information here as well for compat package?

Comment (developer-guy) [reply kubernetes-dns-node-cache-1.22.yaml:41]: also it seems that it should be `kubernetes-dns-node-cache-1.22-compat`, not `kubernetes-csi-node-driver-registrar-compat`

File: kubernetes-dns-node-cache-1.22.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    description: "Compatibility package to place binaries in the location expected by upstream helm charts"
+    pipeline:
+      - runs: |
+          # The helm chart expects the binaries to be in / instead of /usr/bin
+          mkdir -p "${{targets.subpkgdir}}"
+          ln -sf /usr/bin/csi-node-driver-registrar ${{targets.subpkgdir}}/csi-node-driver-registrar
```
Comment (developer-guy) [inline kubernetes-dns-node-cache-1.22.yaml:47]: same issue here

**Comment by developer-guy** [comment]
I think we should ignore the lint issue for package update since we set this version on purpose.

**Review by developer-guy — DISMISSED** [review]

File: kubernetes-dns-node-cache-1.22.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package:
+  name: kubernetes-dns-node-cache-1.22
```
Comment (rawlingsj) [inline kubernetes-dns-node-cache-1.22.yaml:2]:
```suggestion
  name: kubernetes-dns-node-cache
```

File: kubernetes-dns-node-cache-1.22.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      ldflags: -s -w -X github.com/kubernetes/dns/pkg/version.Version=v${{package.version}}
+
+  - uses: strip
+
+update:
+  enabled: true
+  github:
+    strip-prefix: v
+    identifier: kubernetes/dns
+    tag-filter: 1.22.
```
Comment (rawlingsj) [inline kubernetes-dns-node-cache-1.22.yaml:45]:
```suggestion

```

**Review by rawlingsj — DISMISSED** [review]
kubernetes-dns-node-cache doesn't track the Kubernetes versions so AFAIK we don't need to have a version stream for this

**Review by imjasonh — APPROVED** [review]

**Pull request merged**